### DISCUSSION TITLE
Data visualization (stress, strain, etc) for each grain

### DISCUSSION THREAD

**Discussion by shahul31** [neperfepx/neper#471] [Q&A] [answered]
Hello,

Could you please help! Is it possible to visualize the postprocessing results for each grain instead of elementwise, for example - data postprocessing of stress or strain is generated elementwise where the color scheme varies for stress or strain variation, can the same applied for grainwise instead (like given for grain diameter (https://neper.info/tutorials/visualize_tessellation.html#scene-settings). It would be useful to analyze the results for each grain.

**Comment by rquey** [comment, marked as answer]
Assuming that you have a result defined at elements, `<res_name>`, you can compute the grain-average values using `-reselset <res_name>` in `-S`, which will add them to the simulation directory (under `results/elsets/<res_name>`). You will then be able to color the elsets based on this value using `-dataelsetcol real:<res_name>` in `-V` (after loading the simulation directory as input).

**Reply by shahul31** [reply]
Thank you for your answer. I have tried `neper -S Solve.sim -reselset strain33` and other components also, but all the files generated are empty and respective `elts` and `nodes` folders is filled properly. Only this newly generated `elsets` gives out empty files. Could you please help!

**Reply by rquey** [reply]
Make sure you are using the last version, currently 4.4.2-16.

**Reply by shahul31** [reply]
My version is 4.4.2-4, I will update and check again. Thank you very much

**Reply by shahul31** [reply]
It works. Thanks a lot
